### DISCUSSION TITLE
select include

### DISCUSSION THREAD

**Discussion by buzai** [prisma/prisma#23451] [Q&A]
### Question

when relation is one to one, I want the result is Flat, not embedded, how to do? 
eg:
```
i wan result is 

Post {
  id: 1,
  profileUserName: "bob"
}

Instead of

Post {
  id: 1,
  profile: {
     user: { name: "bob" }
  }
}

```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @buzai 👋 

Thank you for raising this question.

Currently, Prisma Client does not support flattening nested result data directly in the query. However, we do have a feature request for it.

- https://github.com/prisma/prisma/issues/9719

The current workaround for this is to manually flatten the result in your application code after you receive the data. For example:

```js
const result = await prisma.post.findUnique({
  where: {
    id: postId,
  },
  include: {
    profile: {
      include: {
        user: true,
      },
    },
  },
});

const flattenedResult = {
  id: result.id,
  profileUserName: result.profile.user.name,
};
```

In this example, flattenedResult will have the structure you want:
```
{
  id: 1,
  profileUserName: "bob"
}
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
